File: core/src/main/java/dev/bee/kanjianki/core/study/InkStroke.java
package dev.bee.kanjianki.core.study;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public final class InkStroke {
    public final List<InkPoint> points;

    public InkStroke(List<InkPoint> points) {
        this.points = Collections.unmodifiableList(new ArrayList<>(points));
    }

    public boolean isEmpty() {
        return points.isEmpty();
    }

    public InkPoint start() {
        return points.isEmpty() ? null : points.get(0);
    }

    public InkPoint end() {
        return points.isEmpty() ? null : points.get(points.size() - 1);
    }
}

File: writing-core/src/main/kotlin/dev/bee/kanjianki/core/study/InkStroke.kt
package dev.bee.kanjianki.core.study

import java.util.ArrayList
import java.util.Collections

class InkStroke(points: List<InkPoint>) {
    @JvmField val points: List<InkPoint> = Collections.unmodifiableList(ArrayList(points))

    fun isEmpty(): Boolean = points.isEmpty()

    fun start(): InkPoint? = if (points.isEmpty()) null else points[0]

    fun end(): InkPoint? = if (points.isEmpty()) null else points[points.size - 1]
}
